pkg/storage/etcd3/watcher.go

```diff
@@ -257,23 +259,30 @@ func (wc *watchChan) transform(e *event) (res *watch.Event) {
 			Object: curObj,
 		}
 	default:
-		curObjPasses := wc.filter(curObj)
-		oldObjPasses := wc.filter(oldObj)
-		switch {
-		case curObjPasses && oldObjPasses:
+		if wc.filterAcceptsAll {
 			res = &watch.Event{
 				Type:   watch.Modified,
 				Object: curObj,
 			}
-		case curObjPasses && !oldObjPasses:
```

Comment:
i feel we have too many indentations here. can we return directly in this case and remove the else at line 267?